infra/discovery/src/listener/listener.js

```diff
@@ -134,84 +136,92 @@ async function main() {
   do {
     start = new Date()
 
-    // Compute the range of blocks to process,
-    // while respecting trailing block configuration.
+    // Compute the range of blocks to process
     const currentBlock = await withRetrys(async () => {
       return context.web3.eth.getBlockNumber()
     })
-    const toBlock = Math.max(currentBlock - context.config.trailBlocks, 0)
+    const toBlock = Math.max(currentBlock, 0)
```

Comment:
Is this needed ? would context.web3.eth.getBlockNumber() ever return a negative block number ?